src/device-state/index.ts

```diff
@@ -364,18 +359,34 @@ export async function getCurrentForReport(
 ): Promise<DeviceState> {
 	const apps = await applicationManager.getState();
 
+	const { apps: targetApps, rejections } =
+		await applicationManager.getTargetAppsWithRejections();
+	const targetAppUuids = Object.keys(targetApps);
+
 	// Fiter current apps by the target state as the supervisor cannot
 	// report on apps for which it doesn't have API permissions
-	const targetAppUuids = Object.keys(await applicationManager.getTargetApps());
-	const appsForReport = Object.keys(apps)
-		.filter((appUuid) => targetAppUuids.includes(appUuid))
-		.reduce(
-			(filteredApps, appUuid) => ({
-				...filteredApps,
-				[appUuid]: apps[appUuid],
-			}),
-			{} as { [appUuid: string]: AppState },
-		);
+	// this step also adds rejected commits for the report
+	const appsForReport = Object.fromEntries(
+		Object.entries(apps).flatMap(([appUuid, app]) => {
+			if (!targetAppUuids.includes(appUuid)) {
+				return [];
+			}
+
+			for (const r of rejections) {
+				if (r.appUuid !== appUuid) {
+					continue;
+				}
+
+				// Add the rejected release to apps for report
+				app.releases[r.releaseUuid] = {
+					update_status: 'rejected',
+					services: {},
+				};
+			}
+
+			return [[appUuid, app]];
```

Comment:
This is another case where I think doing `appsForReport[appUuid] = app` would be far clearer as to the actual intent rather than the current unintentional obfuscation that's happening vs returning nested arrays